modules/caddyhttp/caddyhttp.go

```diff
@@ -297,8 +301,16 @@ func (app *App) automaticHTTPS() error {
 			// tell the server to use TLS by specifying a TLS
 			// connection policy (which supports HTTP/2 and the
 			// TLS-ALPN ACME challenge as well)
-			srv.TLSConnPolicies = caddytls.ConnectionPolicies{
-				{ALPN: defaultALPN},
+			if srv.TLSConnPolicies == nil {
+				// build a new TLSConnPolicies if it does not exist
+				srv.TLSConnPolicies = caddytls.ConnectionPolicies{
+					&caddytls.ConnectionPolicy{ALPN: defaultALPN},
+				}
+			} else {
+				// append the default policy at the end of the existing policies
+				srv.TLSConnPolicies = append(srv.TLSConnPolicies,
+					&caddytls.ConnectionPolicy{ALPN: defaultALPN},
+				)
```

Comment:
@alexandrestein Just a quick question -- does appending the default policy have any particular utility? i.e. is there something that breaks if you don't do this?